crates/red_knot_python_semantic/src/semantic_index/use_def/bitset.rs

```diff
@@ -0,0 +1,402 @@
+use std::collections::{btree_set, BTreeSet};
+
+/// Ordered set of `u32`.
+///
+/// Uses an inline bit-set for small values (up to 64 * B), falls back to a [`BTreeSet`] if a
+/// larger value is inserted.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(super) enum BitSet<const B: usize> {
+    /// Bit-set (in 64-bit blocks) for the first 64 * B entries.
+    Inline([u64; B]),
+
+    /// Overflow beyond 64 * B.
+    Heap(BTreeSet<u32>),
```

Comment:
The semantics of the overall type are ordered-set semantics: a u32 should only appear in it once, and it should stay ordered. These semantics are natural for a bitset representation.

The current design here is that we use a stack-allocated bitset for small values, and fall back to a `BTreeSet` of the u32 values if we overflow the size of the stack-allocated bitset, because a `BTreeSet` has the same ordered-set semantics that we need.

It sounds like you are suggesting an alternative design where we just have a growable bitset instead, using `SmallVec<u64>` to just extend the bitset as far as needed. That sounds like it could also work, and would probably be a bit more efficient when we do need to overflow. I can give it a try.

Reply:
This was a good suggestion, thanks! Doesn't seem to have a significant impact on perf either way, but the code is simpler.

Reply:
Did you made the changes locally? Because I don't see them here yet :D